Sources/StreamChatUI/CommonViews/AvatarView/ChatChannelAvatarView.swift

```diff
@@ -30,31 +47,293 @@ open class ChatChannelAvatarView: _View, ThemeProvider, SwiftUIRepresentable {
+        if let channelAvatarUrl = channel.imageURL {
+            loadChannelAvatar(from: channelAvatarUrl)
+            return
+        }
+      
+        // Use the appropriate method to load avatar based on channel type
+        if channel.isDirectMessageChannel {
+            loadDirectMessageChannelAvatar(channel: channel)
+        } else {
+            loadMergedAvatars(channel: channel)
+        }
+    }
+    
+    /// Loads the avatar from the URL. This function is used when the channel has a non-nil `imageURL`
+    /// - Parameter url: The `imageURL` of the channel
+    open func loadChannelAvatar(from url: URL) {
         presenceAvatarView.avatarView.imageView.loadImage(
-            from: avatarURL,
+            from: url,
             placeholder: appearance.images.userAvatarPlaceholder4,
             preferredSize: .avatarThumbnailSize,
             components: components
         )
+    }
+    
+    /// Loads avatar for a directMessageChannel
+    /// - Parameter channel: The channel
+    open func loadDirectMessageChannelAvatar(channel: ChatChannel) {
+        let lastActiveMembers = channel.lastActiveMembers
+            .sorted { $0.memberCreatedAt < $1.memberCreatedAt }
+            .filter { $0.id != content.currentUserId }
+        
+        // If there are no members other than the current user in the channel, load a placeholder
+        if lastActiveMembers.isEmpty {
+            presenceAvatarView.avatarView.imageView.loadImage(
+                from: nil,
+                placeholder: appearance.images.userAvatarPlaceholder4,
+                preferredSize: .avatarThumbnailSize,
+                components: components
+            )
+            return
+        }
+        
+        let otherMember = lastActiveMembers.first
+        let isOnlineIndicatorVisible = otherMember?.isOnline ?? false
+        
+        presenceAvatarView.avatarView.imageView.loadImage(
+            from: otherMember?.imageURL,
+            placeholder: appearance.images.userAvatarPlaceholder3,
+            preferredSize: .avatarThumbnailSize,
+            components: components
+        )
+        
         presenceAvatarView.isOnlineIndicatorVisible = isOnlineIndicatorVisible
     }
+    
+    /// Loads an avatar which is merged (tiled) version of the first four active members of the channel
+    /// - Parameter channel: The channel
+    open func loadMergedAvatars(channel: ChatChannel) {
+        // The channel is a non-DM channel, hide the online indicator
+        presenceAvatarView.isOnlineIndicatorVisible = false
+        
+        let lastActiveMembers = channel.lastActiveMembers
+            .sorted { $0.memberCreatedAt < $1.memberCreatedAt }
+            .filter { $0.id != content.currentUserId }
+        
+        // If there are no members other than the current user in the channel, load a placeholder
+        if lastActiveMembers.isEmpty {
+            presenceAvatarView.avatarView.imageView.loadImage(
```

Comment:
this piece of code also can be made an private extension for ImageView

```
   presenceAvatarView.avatarView.imageView.loadImage(
                from: nil,
                placeholder: appearance.images.userAvatarPlaceholder4,
                preferredSize: .avatarThumbnailSize,
                components: components
            )
```

There is a 3 places where only name of placeholder changes,  .avatarThumbnailSize and other parameters stay the same, so this can be DRY-out

Reply:
this is annoying if you want to make your own avatar view and dont have your own image loader